pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -199,19 +228,26 @@ def publish(self, topic, data, **attrs):
 
         # Create the Pub/Sub message object.
         message = types.PubsubMessage(data=data, attributes=attrs)
-        if message.ByteSize() > self.batch_settings.max_bytes:
-            raise ValueError(
-                'Message being published is too large for the '
-                'batch settings with max bytes {}.'.
-                format(self.batch_settings.max_bytes)
-            )
 
         # Delegate the publishing to the batch.
         batch = self.batch(topic)
-        future = None
-        while future is None:
-            future = batch.publish(message)
-            if future is None:
-                batch = self.batch(topic, create=True)
+        message_size = message.ByteSize()
+
+        if message_size > batch.settings.max_bytes:
+            with self._batch_lock:
+                backup = self._batches[topic]
+                settings = types.BatchSettings(message_size, 0.05, 1)
+                batch = self._batch(
+                    topic, create=True, batch_settings=settings)
+                future = None
+                while future is None:
+                    future = batch.publish(message)
+                self._batches[topic] = backup
```

Comment:
Rather than save - restore any existing batch for the topic, jjust create a new one, e.g.:

```python
       if message_size > batch.settings.max_bytes:
           with self._batch_lock:
               settings = types.BatchSettings(message_size, 0.05, 1)
               batch = self._batch_class(
                     autocommit=autocommit,
                     client=self,
                     settings=batch_settings,
                     topic=topic,
                 )
               future = None
               while future is None:
                   future = batch.publish(message)
```

We could then remove the whole `_batch()` vs. `batch()` bit altogether.

Reply:
makes sense.